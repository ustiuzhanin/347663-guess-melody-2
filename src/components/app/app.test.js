import React from "react";
import {App} from "./app.jsx";
import renderer from "react-test-renderer";

test(`App's snapshot`, () => {
  const tree = renderer
    .create(
        <App
          settings={{gameTime: 3, maxErrors: 3}}
          questions={[`0`]}
          errorCount={2}
          onUserAnswer={jest.fn()}
          resetProgress={jest.fn()}
          onWelcomeScreenClick={jest.fn()}
          step={-1}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
