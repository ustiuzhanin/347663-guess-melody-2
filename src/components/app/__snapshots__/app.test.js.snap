// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App's snapshot 1`] = `
<section
  className="game"
>
  <header
    className="game__header"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="welcome"
  >
    <div
      className="welcome__logo"
    >
      <img
        alt="Угадай мелодию"
        height="83"
        src="img/melody-logo.png"
        width="186"
      />
    </div>
    <button
      className="welcome__button"
      onClick={[MockFunction]}
    >
      <span
        className="visually-hidden"
      >
        Начать игру
      </span>
    </button>
    <h2
      className="welcome__rules-title"
    >
      Правила игры
    </h2>
    <p
      className="welcome__text"
    >
      Правила просты:
    </p>
    <ul
      className="welcome__rules-list"
    >
      <li>
        За 
        3
         минут нужно ответить на все вопросы.
      </li>
      <li>
        Можно допустить 
        3
         ошибки.
      </li>
    </ul>
    <p
      className="welcome__text"
    >
      Удачи!
    </p>
  </section>
</section>
`;
